### PR TITLE
Update Brewfile for xcodegen, Xcode 14.1

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description

- Add `xcodegen` to Brewfile, required tool for `make validate-spm-build` command

### Checklist
- [n/a] Update `CHANGELOG`

### CODE REVIEW

**Review by ahmedaly16 — APPROVED** [review]

**Review by eugenes78 — APPROVED** [review]
LGTM

File: Brewfile.lock.json [hunk 1]
```diff
@@ -95,45 +95,84 @@
         }
       },
       "swiftformat": {
-        "version": "0.52.11",
+        "version": "0.53.0",
```
Comment (kried) [inline Brewfile.lock.json:98]: Are there any places in the project where the formatting should have changed due to the formatter update?

File: Brewfile.lock.json [hunk 2]
```diff
@@ -143,11 +182,11 @@
   "system": {
     "macos": {
       "ventura": {
-        "HOMEBREW_VERSION": "4.2.2",
+        "HOMEBREW_VERSION": "4.2.5",
         "HOMEBREW_PREFIX": "/opt/homebrew",
         "Homebrew/homebrew-core": "api",
         "CLT": "15.1.0.0.1.1700200546",
-        "Xcode": "14.3.1",
+        "Xcode": "14.1",
```
Comment (kried) [inline Brewfile.lock.json:189]: What are the reasons to downgrade Xcode version? 
Xcodegen claims to be compatible with Xcode 14.3 starting from [2.36.0](https://github.com/yonaskolb/XcodeGen/blob/master/CHANGELOG.md#2360), the project uses 2.38.0

**Review by kried — APPROVED** [review]

File: Brewfile.lock.json (hunk 1 above)
Comment (aokj4ck) [reply Brewfile.lock.json:98]: Oh yes actually 4 total; this adds one newline and changes 3 `.forEach` instances into `for` loops

File: Brewfile.lock.json (hunk 2 above)
Comment (aokj4ck) [reply Brewfile.lock.json:189]: I've just been using Xcode 14.1 everywhere to make sure I don't miss anything on the minimum supported version

**Pull request merged**